app/components/Views/SendFlow/Amount/index.js

```diff
@@ -1192,10 +1234,48 @@ class Amount extends PureComponent {
       internalPrimaryCurrencyIsCrypto,
       currentBalance,
     } = this.state;
-    const { currentCurrency } = this.props;
+    const {
+      currentCurrency,
+      selectedAsset,
+      navigation,
+      isNetworkBuyNativeTokenSupported,
+      swapsIsLive,
+      chainId,
+    } = this.props;
     const colors = this.context.colors || mockTheme.colors;
     const themeAppearance = this.context.themeAppearance || 'light';
     const styles = createStyles(colors);
+    const navigateToSwap = () => {
+      navigation.replace('Swaps', {
+        screen: 'SwapsAmountView',
+        params: {
+          sourceToken: swapsUtils.NATIVE_SWAPS_TOKEN_ADDRESS,
+          destinationToken: selectedAsset.address,
+        },
+      });
+    };
+
+    const navigateToBuyOrSwaps = () => {
+      if (
+        !selectedAsset.isETH &&
+        AppConstants.SWAPS.ACTIVE &&
+        swapsIsLive &&
+        isSwapsAllowed(chainId) &&
+        amountError === strings('transaction.insufficient')
+      ) {
```

Comment:
Q: same as for the app/components/UI/Swaps/index.js useEffect question. Find a way to clarify the test conditions. For instance use a variable `isToken` and `isSwapPossible` and `isFundingNeeded` . Then you have this conditions:
```suggestion
      if (
        isToken
        && isSwapPossible
        && isFundingNeeded
      ) {
```
That may be more readable.

Reply:
Thanks for the review @NicolasMassart definitely addressed it!